From: liuq <liuq131@chinatelecom.cn>
Subject: mm/sparse: remove redundant judgments from macro for_each_present_section_nr
Date: Fri, 7 Jul 2023 14:05:01 +0800

next_present_section_nr() has already ensured that
'section_nr<=__highest_present_section_nr', so this check is removed.

Link: https://lkml.kernel.org/r/20230707060501.29184-1-liuq131@chinatelecom.cn
Signed-off-by: liuq <liuq131@chinatelecom.cn>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/sparse.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mm/sparse.c~mm-sparse-remove-redundant-judgments-from-macro-for_each_present_section_nr
+++ a/mm/sparse.c
@@ -172,8 +172,7 @@ static void __section_mark_present(struc
 
 #define for_each_present_section_nr(start, section_nr)		\
 	for (section_nr = next_present_section_nr(start-1);	\
-	     ((section_nr != -1) &&				\
-	      (section_nr <= __highest_present_section_nr));	\
+	     section_nr != -1;								\
 	     section_nr = next_present_section_nr(section_nr))
 
 static inline unsigned long first_present_section_nr(void)
_
